### DISCUSSION TITLE
Error: Cannot Bridge more than 149 Accessories.

### DISCUSSION THREAD

**Discussion by adey** [iklein99/homebridge-smartthings#126] [Q&A]
Running in to this error. 149 accessories, is this including or excluding the device ignore list?

Thank you.

**Comment by iklein99** [comment]
Looks like an error from Home Kit.  I'm not familiar, but it should not include the excluded devices since they aren't registered with homebridge.

**Reply by adey** [reply]
Sorry, I didn't post the entire error string. Seems like that error is being thrown from getOnlineDevices function in platform.ts at line 123 because the leading text is the same.

      `}).catch(error => {
        this.log.error('Error getting devices from Smartthings: ' + error);
        reject();
      });`

Devices that were cached and subsequently added to the ignore devices list, do they then get removed from the cache? This may be the root cause why even when you add enough devices to the ignore list to bring the list down below 149 accessories the error still persists.

Thanks.

**Comment by iklein99** [comment]
When you restart homebridge, the plugin will first get a list of all devices from homebridge, then will attempt to register them, ignoring the ones that are in the ignore list, then it will unregister any devices that weren’t registered. 

So the problem is probably that it can’t register any more devices since the cache is full.

You can “start over” by deleting ~/.homebridge/accessories/cachedAccessories

**Reply by phuonglho1968** [reply]
Hi,

I have a lots of devices in ST like 245+ devices.  Can you please increase the limit that allow to import into homebit.

Thanks!

**Reply by ineedhungry** [reply]
This is a limit of homekit.  149 per bridge.  Really we just need a way to limit imported devices, or break it up into multiple instances.
